packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -559,45 +577,60 @@ class HeroController extends NavigatorObserver {
   }
 
   @override
-  void didStartUserGesture() {
-    _questsEnabled = false;
-  }
-
-  @override
-  void didStopUserGesture() {
-    _questsEnabled = true;
+  void didStartUserGesture(Route<dynamic> route, Route<dynamic> previousRoute) {
+    assert(navigator != null);
+    assert(route != null);
+    _maybeStartHeroTransition(route, previousRoute, HeroFlightDirection.pop, true);
   }
 
   // If we're transitioning between different page routes, start a hero transition
   // after the toRoute has been laid out with its animation's value at 1.0.
-  void _maybeStartHeroTransition(Route<dynamic> fromRoute, Route<dynamic> toRoute, HeroFlightDirection flightType) {
-    if (_questsEnabled && toRoute != fromRoute && toRoute is PageRoute<dynamic> && fromRoute is PageRoute<dynamic>) {
+  void _maybeStartHeroTransition(
+    Route<dynamic> fromRoute,
+    Route<dynamic> toRoute,
+    HeroFlightDirection flightType,
+    [bool fromGesture = false]
```

Comment:
I don't think this needs to be an optional parameter since it's private and only called in 2 places and you have to specify it in one of them.

The parameter name sounds like a gesture descriptor. Maybe isUserPopTransition.

Reply:
Removed optional. Though I kept it the same name as above (to reduce cognitive load for future readers)